src/javascripts/components/copy.mjs

```diff
@@ -19,34 +20,50 @@ class Copy {
 
     this.$button = document.createElement('button')
     this.$button.className = 'app-copy-button'
-    this.$button.setAttribute('aria-live', 'assertive')
     this.$button.textContent = 'Copy code'
 
+    this.$status = document.createElement('span')
+    this.$status.className = 'govuk-visually-hidden'
+    this.$status.setAttribute('aria-live', 'assertive')
+
+    this.$module.insertAdjacentElement('beforebegin', this.$status)
     this.$module.insertAdjacentElement('beforebegin', this.$button)
-    this.copyAction()
+
+    const $clipboard = new ClipboardJS(this.$button, {
+      target: (trigger) => trigger.nextElementSibling
+    })
+
+    $clipboard.on('success', (event) => this.successAction(event))
+    $clipboard.on('error', (event) => this.resetAction(event))
   }
 
   /**
-   * Set up button copy action
+  * Copy to clipboard success
+  *
+  * @param {import('clipboard').Event} event
+  */
+  successAction(event) {
+    this.$button.textContent = this.$status.textContent = 'Code copied'
+      // Reset button after 5 seconds
+    this.resetAction(event, 5000)
+  }
+
+  /**
+   * Copy to clipboard reset
+   *
+   * @param {import('clipboard').Event} event
+   * @param {number} [timeout] - Button text reset timeout
    */
-  copyAction() {
-    // Copy to clipboard
-    try {
-      new ClipboardJS(this.$button, {
-        target: (trigger) => trigger.nextElementSibling
-      }).on('success', (event) => {
-        this.$button.textContent = 'Code copied'
-        event.clearSelection()
-        setTimeout(() => {
-          this.$button.textContent = 'Copy code'
-        }, 5000)
-      })
-    } catch (error) {
-      if (error instanceof Error) {
-        console.log(error.message)
-      }
-    }
+  resetAction(event, timeout = 0) {
+    event.clearSelection()
+
+    // Reset button after timeout
+    setTimeout(() => {
```

Comment:
Testing complete, looks brill except for a bug brought over from the previous version

Hammering the "Copy code" button creates many 5 second timers 😮

Quick one to fix whilst we're here?

```suggestion
    if (this.resetTimeoutId) {
      clearTimeout(this.resetTimeoutId)
    }

    // Reset button after timeout
    this.resetTimeoutId = setTimeout(() => {
```